week-1-ts-ecosystem/w01e04-light-me-up/src/task-2.ts

```diff
@@ -0,0 +1,41 @@
+export class PowerSource {
+    private energySupply = 100
+  
+    consume(energy: number): void {
+      try {
+        if(this.energySupply >= energy) {
+          this.energySupply -= energy
+        } else {
+          throw new Error(`Not enough energy. Current energy supply: ${this.energySupply}. Your device requires: ${energy}`);
+        }
+      } catch (error) {
+        console.error(error)
+      }
+    }
+  }
+  
+  class LightBulb {
+    public readonly powerConsumption = 20
+
+    constructor(private source: PowerSource) {}
```

Comment:
Tutaj jest dobrze przekazane źródło mocy i wykorzystane.

ale dalej do poprawki obsługa błędu - której tutaj nie widzę

Reply:
Dodatkowo `LightBulb` logika się powtarza - to powinno mieć osobny plik dedykowany dla klasy.